### PR TITLE
Form submission notification text translation

### PR DESCRIPTION
**Pull request by stephherbers**
## Product Description

[USH-3861](https://dimagi-dev.atlassian.net/browse/USH-3861) and [USH-3885](https://dimagi-dev.atlassian.net/browse/USH-3885
)
Model changes [here](https://github.com/dimagi/commcare-hq/pull/33881)

image

## Technical Summary

This is just for one of the model changes, the other is in a separate PR
### Safety story

The field will still not be added if it is blank, and now with no migration it will be blank until the field is manually added and uploaded by someone with access.

### Automated test coverage

There are some in ap strings and I added another one

### QA Plan

I am not requesting QA

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

[USH-3861]: https://dimagi-dev.atlassian.net/browse/USH-3861?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[USH-3885]: https://dimagi-dev.atlassian.net/browse/USH-3885?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: corehq/apps/app_manager/models.py [hunk 1]
```diff
@@ -1347,6 +1347,11 @@ def get_submit_label(self, lang):
             return self.submit_label[lang]
         return 'Submit'
 
+    def get_submit_notification_label(self, lang):
+        if lang in self.submit_notification_label:
```
Comment (nospame) [inline corehq/apps/app_manager/models.py:1351]: Does this work if `self.submit_notification_label` doesn't exist yet? Or maybe it's already known to exist with its default value based on the model change -- not certain how that works.

Comment (stephherbers) [reply corehq/apps/app_manager/models.py]: good call added check https://github.com/dimagi/commcare-hq/pull/33885#:~:text=in%20getter%20function-,9a288ed,-Merge%20state

**Review by AddisonDunn — CHANGES_REQUESTED** [review]

File: corehq/apps/app_manager/models.py [hunk 2]
```diff
@@ -1347,6 +1347,11 @@ def get_submit_label(self, lang):
             return self.submit_label[lang]
         return 'Submit'
 
+    def get_submit_notification_label(self, lang):
+        if self.submit_notification_label and lang in self.submit_notification_label:
+            return self.get_submit_notification_label[lang]
```
[the remark this answers is not in the document]
Comment (stephherbers) [reply corehq/apps/app_manager/models.py]: https://github.com/dimagi/commcare-hq/pull/33885#:~:text=return%20submit_notification_label%5Blang%5D-,971e16a,-allow%20for%20blank

**Review by AddisonDunn — COMMENTED** [review]
Screenshot 2023-12-19 at 2 19 45 PM
Screenshot 2023-12-19 at 2 45 48 PM

Looking good on staging. Besides that one bit of logic can be extracted, everything looks in order to me.

**Comment by stephherbers** [comment]
@AddisonDunn cool for the sake of time with the deploy deadline I'll plan on following up on the nits at a later date

**Review by AddisonDunn — APPROVED** [review]

File: corehq/apps/translations/app_translations/download.py [hunk 3]
```diff
@@ -381,8 +381,10 @@ def get_form_question_label_name_media(langs, form):
                 itext_items[text_id][(lang, value_form)] = value
 
     itext_items['submit_label'] = {}
+    itext_items['submit_notification_label'] = {}
     for lang in langs:
         itext_items['submit_label'][(lang, 'default')] = form.get_submit_label(lang)
+        itext_items['submit_notification_label'][(lang, 'default')] = form.get_submit_notification_label(lang)
```
Comment (esoergel) [inline corehq/apps/translations/app_translations/download.py:387]: we spoke about this offline, but I think it'd make sense to move these two to a separate helper function, as this one deals with form questions.  That's also allow for dropping the "allow empty for submit_notification_label row" stuff.

**Pull request merged**